### PR TITLE
Part 1: Sign In Jetpack basic check and error messaging

### PR DESCRIPTION
**Pull request by AmandaRiu**
This PR partially addresses #1091. To keep the PRs smaller, I'm breaking them down for review to be merged into a feature branch. This PR covers the following changes:
- Removes any mention of Jetpack from the login prologue
- Starts the login with the site address screen
- Updates label text for `LoginEmailFragment` to use the site url
- Updates error messaging in `LoginSiteAddressFragment` for all scenarios except the "not connected" scenario (coming in a later PR)
- Uses `SiteAction.FETCH_CONNECT_SITE_INFO` to get information on the URL entered to determine if the site has WordPress installed, and if it has Jetpack installed and connected. 
- Adds the new "Jetpack Required" fragment (help buttons work, but other buttons just show toast message at this point).

**Note**: The code to check the URL for WordPress and Jetpack is a temporary measure for this round. If we decide to fully implement this login flow, we'll need to use the full XMLRPC discovery code and implement handlers for various scenarios such as HTTP Authentication required and self-signed SSL certifications.

## Not a WordPress site
The initial designs required a check for WooCommerce on a site, but this is not possible without a major overhaul. Instead we check if WordPress is installed, and if not, we display an error:

## Jetpack not connected
If jetpack is not installed **and** connected properly, the user will be directed to the "Jetpack Required" view:

## WooCommerce site
If the site is a WordPress site and has Jetpack installed and connected, the user should be routed to login by email as normal. Other than the label changes, everything else should remain the same for the purpose of this PR:

cc: @kyleaparker for design review

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml [hunk 1]
```diff
@@ -1,100 +1,84 @@
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-                xmlns:app="http://schemas.android.com/apk/res-auto"
-                android:layout_width="match_parent"
-                android:layout_height="match_parent"
-                android:background="@color/white">
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (anitaa1990) [inline WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml]: I love that you migrated everything to `ConstraintLayout` 😄

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginSiteAddressFragment.java [hunk 2]
```diff
@@ -162,7 +166,15 @@ protected void discover() {
         mRequestedSiteAddress = cleanedSiteAddress;
 
         String cleanedXmlrpcSuffix = UrlUtils.removeXmlrpcSuffix(mRequestedSiteAddress);
-        mDispatcher.dispatch(SiteActionBuilder.newFetchWpcomSiteByUrlAction(cleanedXmlrpcSuffix));
+
+        if (mLoginListener.getLoginMode() == LoginMode.WOO_LOGIN_MODE) {
```
Comment (anitaa1990) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginSiteAddressFragment.java]: As I understand, we are using `WOO_LOGIN_MODE` to get the connection info of the site and then based on the response, either display `Site cannot be connected` error, `Not a WordPress site` error or `Jetpack not installed` error. If Jetpack is installed, we just redirect to the `email` screen and if not, then redirect to the Jetpack needed screen. Is my understanding correct?

**Review by anitaa1990 — COMMENTED** [review]
This looks great @AmandaRiu! I love how you migrated a lot of the UI to `ConstraintLayout` 😄 

I tested the functionality and everything works good 👍 I am still trying to understand how the login lib works so not sure if I am the best person to review 😊

File: WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
+package com.woocommerce.android
```
Comment (nbradbury) [inline WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt]: Is there a reason this isn't in the `ui/login` folder?

Screen Shot 2019-06-04 at 12 36 07 PM

File: WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        setHasOptionsMenu(true)
+
+        // Inflate the layout for this fragment
```
Comment (nbradbury) [inline WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt]: I know this is nitpicky (Nickpicky?), but is this comment necessary?

**Comment by nbradbury** [comment]
When I tap "Need more help" here:

Screenshot_1559670289

It immediately asks me for my name and email address:

Screenshot_1559670292

I think it should simply show the Help & Support activity, just as it does when the Help button is tapped in the toolbar.

**Comment by nbradbury** [comment]
@AmandaRiu I'm finished with my first round and it looks good despite my nitpicks. However, one potentially big problem I'm seeing is that after testing with a site that had Jetpack disabled, I then enabled Jetpack and tried again but continued to see the Jetpack required screen. 

I'm assuming this is a caching issue, or something similar, but it could be a annoyance for users that enable Jetpack and still can't login.

File: WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml (hunk 1 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml]: That's my jam. _**LOVE**_ ConstraintLayout!

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginSiteAddressFragment.java (hunk 2 above)
Comment (AmandaRiu) [reply libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginSiteAddressFragment.java]: That's correct!

File: WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt (hunk 3 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt]: I'm not sure what happened there. I manually moved it once, and then again in the latest branch I'm working on for the second PR. I've manually moved it again 🤞 a0b19dcc032619cf3763b10cfcaf81af49e92be1

File: WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt (hunk 4 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/java/com/woocommerce/android/LoginJetpackRequiredFragment.kt]: ha! no that's a really good suggestion. Done in fe639d2975fd7f5acd5a923d820398420a557560

**Review by AmandaRiu — COMMENTED** [review]
@nbradbury Thank you for the quick review! I've addressed the comments inline and referenced code changes where needed. 

> I think it should simply show the Help & Support activity, just as it does when the Help button is tapped in the toolbar.

This is pre-existing functionality and I'd prefer not to tinker with it for this project. I'd be down for changing the flow, it would be pretty easy, but I'd definitely like to get @kyleaparker 's thumbs up on it, so we should just create an official ticket so it can be reviewed. 

> I'm assuming this is a caching issue, or something similar, but it could be a annoyance for users that enable Jetpack and still can't login.

Oh that's no good. I'll have to look into that as well. There must be something I can do there. I'll add it to my TODO list for the project. I wouldn't worry about it for this PR as this is focused more on the flow. Even the way we're checking for Jetpack is temporary. 

Ready for another round!

**Comment by nbradbury** [comment]
@AmandaRiu I was about to merge this then thought I should test against a couple more sites. Both sites I tested with end up showing that Jetpack screen, even though both of them are wp.com sites running WooCommerce.

I know you mentioned "The code to check the URL for WordPress and Jetpack is a temporary measure for this round" - does that mean go ahead and merge even if it's not working?

Aside: I'm assuming this is why my self-hosted site kept showing that screen even after I enabled Jetpack, so the caching issue I mentioned is likely a red herring.

**Comment by AmandaRiu** [comment]
@nbradbury 

>  I was about to merge this then thought I should test against a couple more sites. Both sites I tested with end up showing that Jetpack screen, even though both of them are wp.com sites running WooCommerce.

After some digging we figured out that Atomic sites have different values returned during the Connected Site check. This is an example response:
```
exists: true
hasJetpack: true
isJetpackActive: false
isJetpackConnected: false
isWordPress: true
isWordPressDotCom: true
jetpackVersion: "7.4"
skipRemoteInstall: false
urlAfterRedirects: "amandariu.com"
```

This is an atomic store and it seems like these values will always be false for an atomic store:
```
isJetpackActive: false
isJetpackConnected: false
```
I tested with a temp ninja site and walked through the results of the connect site check and verified that during each step of jetpack installation process, install, activate, connect...those values update as expected and can be relied on. I've updated the logic to only require `hasJetpack = true` if `isWordPressDotCom = true` in d347c49da9eecdc0f0c29aec215711e18aa4b109. This is now working as expected. Ready for testing!

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Comment by kyleaparker** [comment]
>This is pre-existing functionality and I'd prefer not to tinker with it for this project. I'd be down for changing the flow, it would be pretty easy, but I'd definitely like to get @kyleaparker 's thumbs up on it, so we should just create an official ticket so it can be reviewed.

👍 ok to leave for the first milestone, will take a look and see if there are improvements we could make later on

**Pull request merged**